pkg/volume/configmap/configmap.go

```diff
@@ -190,8 +190,9 @@ func (b *configMapVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 	return nil
 }
 
-func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap) (map[string][]byte, error) {
-	payload := make(map[string][]byte, len(configMap.Data))
+func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap) (payload map[string][]byte, mode map[string]int32, err error) {
```

Comment:
How about we pass the default mode to makePayload, and have it explicitly set this field for every file.